pkg/runtime/scheduler/scheduler.go

```diff
@@ -254,6 +258,13 @@ func (s *Scheduler) connSchedulerHosts(ctx context.Context) (schedulerv1pb.Sched
 		addresses = append(addresses, host.GetAddress())
 	}
 
+	// Let the code above run to this point to validate the connection and WatchHosts RPC call
+	// If we are in standalone mode and only one address is provided, we need to return the address as provided in config
+	// Otherwise the resolved address from response will have the address of the docker network
+	if len(s.addresses) > 0 && s.mode == modes.StandaloneMode {
+		addresses = s.addresses
+	}
```

Comment:
We can't do this- this is going to fail in standalone mode on an existing deployment when the scheduler replicas are scaled up (or down). This is one of the main reasons for this API existing in the first place. Not least, daprd side hacks like these are super fragile and can back us into a corner long term.

If scheduler is listening on localhost and this is inaccessible to the host machine, how is scheduler reachable at all? If it's listening on another interface, then that IP too should be reachable from the host machine and the existing logic in Scheduler to use the right address to broadcast should also work.

If we *must must* do something here, I think it best to add a `--override-broadcast-address` CLI flag on the Scheduler side.